arangod/RocksDBEngine/RocksDBChecksumEnv.cpp

```diff
@@ -0,0 +1,264 @@
+    if (auto it = _fileNamesToHashes.find(baseName);
+        it != _fileNamesToHashes.end()) {
+      checksum = it->second;
+      _fileNamesToHashes.erase(it);
+    }
+  }
+  return checksum;
+}
+
+std::string ChecksumHelper::buildShaFileNameFromSst(
+    std::string const& fileName, std::string const& checksum) {
+  if (!fileName.empty() && !checksum.empty()) {
+    TRI_ASSERT(fileName.size() > 4);
+    std::string shaFileName = fileName.substr(0, fileName.size() - 4);
+    TRI_ASSERT(!isFileNameSst(shaFileName));
+    shaFileName += ".sha." + checksum + ".hash";
+    return shaFileName;
+  }
+  return {};
+}
+
+rocksdb::Status ChecksumWritableFile::Append(const rocksdb::Slice& data) {
+  _checksumCalc.updateIncrementalChecksum(data.data(), data.size());
+  return rocksdb::WritableFileWrapper::Append(data);
+}
+
+rocksdb::Status ChecksumWritableFile::Close() {
+  TRI_ASSERT(_helper != nullptr);
+  _checksumCalc.computeFinalChecksum();
+  if (!_helper->writeShaFile(_fileName, _checksumCalc.getChecksum())) {
+    LOG_TOPIC("0b00e", ERR, arangodb::Logger::ENGINES)
+        << "Writing sha file for " << _fileName << " was unsuccessful";
+  }
+  return rocksdb::WritableFileWrapper::Close();
+}
+
+rocksdb::Status ChecksumEnv::NewWritableFile(
+    const std::string& fileName, std::unique_ptr<rocksdb::WritableFile>* result,
+    const rocksdb::EnvOptions& options) {
+  std::unique_ptr<rocksdb::WritableFile> writableFile;
+  rocksdb::Status s =
+      rocksdb::EnvWrapper::NewWritableFile(fileName, &writableFile, options);
+  if (!s.ok()) {
+    return s;
+  }
+  try {
+    if (_helper->isFileNameSst(fileName)) {
+      *result = std::make_unique<ChecksumWritableFile>(std::move(writableFile),
+                                                       fileName, _helper);
+    } else {
+      *result = std::move(writableFile);
+    }
+  } catch (std::exception const& e) {
+    LOG_TOPIC("8b19e", ERR, arangodb::Logger::ENGINES)
+        << "WritableFile: exception caught when allocating " << e.what();
+    return rocksdb::Status::MemoryLimit(
+        "WritableFile: exception caught when allocating");
+  }
+  return s;
+}
+
+rocksdb::Status ChecksumEnv::DeleteFile(const std::string& fileName) {
+  if (_helper->isFileNameSst(fileName)) {
+    std::string checksum = _helper->removeFromTable(fileName);
+    std::string shaFileName =
+        _helper->buildShaFileNameFromSst(fileName, checksum);
+    if (!shaFileName.empty()) {
+      auto res = TRI_UnlinkFile(shaFileName.c_str());
+      if (res == TRI_ERROR_NO_ERROR) {
+        LOG_TOPIC("e0a0d", DEBUG, arangodb::Logger::ENGINES)
+            << "deleteCalcFile: delete file succeeded for " << shaFileName;
+      } else {
+        LOG_TOPIC("acb34", WARN, arangodb::Logger::ENGINES)
+            << "deleteCalcFile: delete file failed for " << shaFileName;
+      }
+    }
+  }
+  rocksdb::Status res = rocksdb::EnvWrapper::DeleteFile(fileName);
+  if (res == rocksdb::Status::OK()) {
```

Comment:
Can be simplified to just
```suggestion
  if (res.ok()) {
```